### PR TITLE
Add info about last version supporting IE9 and IE10 and remove obsolete code

### PR DESCRIPTION
**Pull request by archmoj**
@plotly/plotly_js

### CODE REVIEW

File: src/lib/loggers.js [hunk 1]
```diff
@@ -82,27 +82,10 @@ loggers.error = function() {
...
-    for(var i = 0; i < args.length; i++) {
-        try {
-            f(args[i]);
-        } catch(e) {
-            // still fails - last resort simple console.log
-            console.log(args[i]);
-        }
+        // `this` should always be console, since here we're always
+        // applying a method of the console object.
+        f.apply(console, args);
```
Comment (alexcjohnson) [inline src/lib/loggers.js:89]: Could remove this function entirely now, just inline the apply calls. Also the `console.trace` fallback `|| console.log` in log and warn [could be removed](https://developer.mozilla.org/en-US/docs/Web/API/console/trace)

File: test/image/strict-d3.js [hunk 2]
```diff
@@ -2,6 +2,9 @@
  * strict-d3: wrap selection.style to prohibit specific incorrect style values
  * that are known to cause problems in IE (at least IE9)
  */
+
+// Do we need this process now that IE9 and IE10 are not supported?
```
Comment (alexcjohnson) [inline test/image/strict-d3.js:6]: yes, if you look down below there's plenty here that applies to other browsers too. There may be some parts we could relax but no harm leaving it all for now.

**Comment by nicolaskruchten** [comment]
Hmm so one last-minute concern I have here is that *we* dropped support in our build, but anyone building themselves without `browserify` might still have IE9/IE10 support, so we shouldn't unwind these browser-specific bits of code if possible.

File: src/lib/index.js [hunk 3]
```diff
@@ -727,6 +722,8 @@ lib.isIOS = function() {
     return IS_IOS_REGEX.test(window.navigator.userAgent);
 };
 
+// Do we need this process now that IE9 and IE10 are not supported?
```
Comment (alexcjohnson) [inline src/lib/index.js:725]: yeah, we could probably get rid of this now - and revert the change here that's its only consumer: https://github.com/plotly/plotly.js/commit/98adfaee3ed79c6f23a7fb8bbe1fe27474229f23#diff-9c325f1fe845cf45c705407012f4d4d0be31a9d9fbead70d1f70c5dedde9c634

**Comment by alexcjohnson** [comment]
> so we shouldn't unwind these browser-specific bits of code if possible.

I disagree - I think it's easier to simply say "we don't support IE<11 anymore," otherwise we're going to keep carrying around arcane practices nobody can remember the rationale for... which of course we have plenty of but each instance of this has adds nonzero ongoing cost. And I really don't want to deal with questions like "OK I know you SAID you don't support these browsers but I found this crazy workaround to still build for them and you broke it with something in version 2.13.0."

File: src/lib/loggers.js (hunk 1 above)
Comment (archmoj) [reply src/lib/loggers.js]: Good call. Done in https://github.com/plotly/plotly.js/pull/5373/commits/bc2b0d112b14db802978c76f3e59aab6e0f87155.

File: test/image/strict-d3.js (hunk 2 above)
Comment (archmoj) [reply test/image/strict-d3.js]: OK. I removed the comment in https://github.com/plotly/plotly.js/pull/5373/commits/9276d08e5298c0900daca36cfa9b150f57b400b5.

File: src/lib/index.js (hunk 3 above)
Comment (archmoj) [reply src/lib/index.js]: Addressed in https://github.com/plotly/plotly.js/pull/5373/commits/42c97e2127384de010e7c3de3c4d42369f03087a.

**Comment by nicolaskruchten** [comment]
OK, I see your point, Alex. I guess I'm worried that some people might be happily using IE9/IE10 (ok, "happily") and then in a minor we drop support for them, which is not semver-friendly. Arguably we already did that in 1.49.5 but that's not really a good reason to double down.

**Comment by alexcjohnson** [comment]
Fair enough. How about we split this PR in two: one we merge right now that just documents what already happened but makes no code changes; and a second that we merge in the upcoming v2 that removes IE9/10 cruft.

**Comment by alexcjohnson** [comment]
And we can include a note:

> 1.49.5 inadvertently dropped IE9/10 support by upgrading browserify, but if you bundle plotly.js yourself using the previous version or a different bundler you may still be able to support these browsers until plotly.js v2.0 explicitly drops IE9/10 support.

**Comment by archmoj** [comment]
> Fair enough. How about we split this PR in two: one we merge right now that just documents what already happened but makes no code changes; and a second that we merge in the upcoming v2 that removes IE9/10 cruft.

I kind of disagree. That could potentially be a blocker for getting #5230 in v1.
Unless one opens a regression ticket for IE9. Then I would be happy to pin down browserify to v16.3.0.
On another note streambed now uses browserify at v17.0.0.

**Comment by archmoj** [comment]
> Unless one opens a regression ticket for IE9. Then I would be happy to pin down browserify to v16.3.0.
> On another note streambed now uses browserify at v17.0.0.

That also collides with #5358.

**Comment by nicolaskruchten** [comment]
I think that we can and should merge a PR right away that only:

1. Removes the IE9 notes from our readmes
2. Notes in the changelog that version 1.49.5 dropped IE9/IE10 support *in our dist bundles/on NPM*, which is all that we know for sure that we did

Let's not add all sorts of additional opportunistic cleanup as part of this otherwise-uncontroversial documentation PR please, and we can discuss all the other v1/v2 stuff tomorrow?
